### PR TITLE
Adding the Surface-model Ontology

### PR DESCRIPTION
**Pull request by jtdemer**
The ontology was created on 02/27/2014 with the latest version of the Ontology Creator and uses the current Excel file as input: Metadata, which was created in collaboration with all data providers from the GAIA-X4PLCC-AAD project and can now be merged into the main branch.

Signed-off-by: jtdemer <johannes.demer@asc-s.de>

### CODE REVIEW

File: surface-model/surfacemodel_ontology.ttl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@prefix gax-core: <https://w3id.org/gaia-x/core#> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix surfacemodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/surfacemodel/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+surfacemodel: a owl:Ontology ;
+    rdfs:label "ontology definition for surfaceModel"@en ;
+    dcterms:contributor "Sebastian Tuttas (3DM)" ;
+    owl:versionInfo "0.1"^^xsd:float .
```
Comment (robertschubert) [inline surface-model/surfacemodel_ontology.ttl:11]: ^^xsd:float can be removed

File: surface-model/surfacemodel_ontology.ttl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+surfacemodel:surfaceModel a owl:Class ;
+    rdfs:label "class definition for surfaceModel" ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_ontology.ttl:14]: i would appreciate to have similiar terms for surfaceModel. 
See line below it is called: "surface model".

File: surface-model/surfacemodel_ontology.ttl [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    rdfs:comment "Attributes for surface model"@en ;
+    rdfs:subClassOf gax-core:Resource .
```
Comment (robertschubert) [inline surface-model/surfacemodel_ontology.ttl:16]: should be gx:DataResource.
Please also add the gx prefix 
`@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .`
to both ontology and SHACL

File: surface-model/surfacemodel_shacl.ttl [hunk 4]
```diff
@@ -0,0 +1,137 @@
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix surfacemodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/surfacemodel/> .
+@prefix general: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/general/> .
+@prefix georeference: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/georeference/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+surfacemodel:surfaceModelShape a sh:NodeShape ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:8]: Classes should be in PascalCase:
`surface-model:SurfaceModelShape a sh:NodeShape ;`

File: surface-model/surfacemodel_ontology.ttl [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix surfacemodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/surfacemodel/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+surfacemodel: a owl:Ontology ;
+    rdfs:label "ontology definition for surfaceModel"@en ;
+    dcterms:contributor "Sebastian Tuttas (3DM)" ;
+    owl:versionInfo "0.1"^^xsd:float .
+
+surfacemodel:surfaceModel a owl:Class ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_ontology.ttl:13]: Class should be in PascalCase 
--> SurfaceModel

File: surface-model/surfacemodel_shacl.ttl [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            sh:minCount 1 ;
+            sh:node surfacemodel:QuantityShape ;
+            sh:order 0 ;
+            sh:path surfacemodel:quantity ],
+        [ sh:maxCount 1 ;
+            sh:minCount 0 ;
+            sh:node surfacemodel:GeoreferenceShape ;
+            sh:order 7 ;
+            sh:path georeference:georeference ] ;
+    sh:targetClass surfacemodel:surfaceModel .
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:34]: after the ontology change it should refer to
`surface-model:SurfaceModel .`

File: surface-model/surfacemodel_shacl.ttl [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+surfacemodel:DataShape a sh:NodeShape ;
+    sh:property [ skos:example "nan" ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:37]: if there is no example then this should be removed. Or give a better example than "nan"

File: surface-model/surfacemodel_shacl.ttl [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+            sh:datatype xsd:string ;
+            sh:description "Height, friction values, grey values"@en ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:39]: looking at the description:
can you ad an sh:in to define the enum of possible values?

File: surface-model/surfacemodel_shacl.ttl [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+surfacemodel:FormatShape a sh:NodeShape ;
+    sh:property [ skos:example "ASAM OpenCRG; DLM" ;
+            sh:datatype xsd:string ;
+            sh:description "Format type definition"@en ;
+            sh:in ("ASAM OpenCRG" "DLM") ;
+            sh:maxCount 1 ;
+            sh:message "Validation of type failed!"@en ;
+            sh:name "type"@en ;
+            sh:order 0 ;
+            sh:path surfacemodel:type ],
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:56]: this is the second time where we have 
`sh:path surfacemodel:type`

didnt you mean `surface-model.format`?

File: surface-model/surfacemodel_shacl.ttl [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+            sh:datatype xsd:string ;
+            sh:description "Height, friction values, grey values"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of type failed!"@en ;
+            sh:name "type"@en ;
+            sh:order 0 ;
+            sh:path surfacemodel:type ] ;
+    sh:targetClass surfacemodel:Data .
+
+surfacemodel:FormatShape a sh:NodeShape ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:47]: is this FormatShape different from the FormatShape defined in the other SHACL shapes?
If no then it should be reused (and not duplicated).

File: surface-model/surfacemodel_shacl.ttl [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+            sh:datatype xsd:string ;
+            sh:description "Version of data format"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of version failed!"@en ;
+            sh:name "version"@en ;
+            sh:order 1 ;
+            sh:path surfacemodel:version ] ;
+    sh:targetClass surfacemodel:Format .
+
+surfacemodel:QualityShape a sh:NodeShape ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:67]: I think I have seen QualityShape and QuantityShape in some other branches. Can this also be reused?

File: surface-model/surfacemodel_shacl.ttl [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+            sh:order 2 ;
+            sh:path surfacemodel:orientation ],
+        [ skos:example "true" ;
+            sh:datatype xsd:boolean ;
+            sh:description "Existence of horizontal starting platform"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of platform failed!"@en ;
+            sh:name "platform"@en ;
+            sh:order 3 ;
+            sh:path surfacemodel:platform ],
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:91]: can you rename it to `platformExists`?

File: surface-model/surfacemodel_shacl.ttl [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+            sh:order 1 ;
+            sh:path surfacemodel:resolutionLateral ],
+        [ skos:example "true" ;
+            sh:datatype xsd:boolean ;
+            sh:description "Existence of smooth ramp from platform to road"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of ramp failed!"@en ;
+            sh:name "ramp"@en ;
+            sh:order 4 ;
+            sh:path surfacemodel:ramp ] ;
```
Comment (robertschubert) [inline surface-model/surfacemodel_shacl.ttl:107]: can you rename it to rampExists?

**Review by robertschubert — CHANGES_REQUESTED** [review]

File: surface-model/surfacemodel_ontology.ttl (hunk 1 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl:11]: fixed

File: surface-model/surfacemodel_ontology.ttl [hunk 14]
```diff
@@ -0,0 +1,17 @@
+@prefix dcterms: <http://purl.org/dc/terms/> .
+@prefix gax-core: <https://w3id.org/gaia-x/core#> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix surfacemodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/surfacemodel/> .
```
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl:5]: replace surfacemodel to surface-model for namespace in ontology and shacl file

File: surface-model/surfacemodel_ontology.ttl (hunk 2 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl:14]: fixed

File: surface-model/surfacemodel_ontology.ttl (hunk 3 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl]: fixed

File: surface-model/surfacemodel_ontology.ttl [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
```
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl:8]: merge main into branch - done

File: surface-model/surfacemodel_ontology.ttl (hunk 14 above)
[the remark this answers is not in the document]
Comment (robertschubert) [reply surface-model/surfacemodel_ontology.ttl]: I would also rename the file to match the directory name.
Currently we have `surfacemodel_ontology.ttl` (along with the shacl).
I would prefer `surface-model_ontology.ttl` and `surface-model_shacl.ttl`

File: surface-model/surfacemodel_ontology.ttl (hunk 15 above)
[the remark this answers is not in the document]
Comment (robertschubert) [reply surface-model/surfacemodel_ontology.ttl:8]: we have an encoding issue in the SHACL file so the VARIABLES.md is not generated. I will have a look.

[the remark this answers is not in the document]
Comment (robertschubert) [reply surface-model/surfacemodel_ontology.ttl:8]: @MircoNierenz 
I changed line 161 `Großraum` to fit to UTF8.
Now the VARIABLES.md got generated.
I will resolve this task.

File: surface-model/surfacemodel_ontology.ttl (hunk 1 above)
Comment (robertschubert) [reply surface-model/surfacemodel_ontology.ttl:12]: it is still there:
image

File: surface-model/surfacemodel_ontology.ttl (hunk 2 above)
Comment (robertschubert) [reply surface-model/surfacemodel_ontology.ttl]: Now we have `surface-model` vs. `surface model` in label and comment. 
I would prefer to have one term.

as 2nd (I missed this yesterday): 
ClassNames should be PascalCase
`surfacemodel:surfaceModel` --> `surfacemodel:SurfaceModel`

File: surface-model/surfacemodel_ontology.ttl (hunk 3 above)
Comment (robertschubert) [reply surface-model/surfacemodel_ontology.ttl]: still unchanged. Did you forget to push?

image

File: surface-model/surfacemodel_shacl.ttl (hunk 12 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_shacl.ttl]: done

File: surface-model/surfacemodel_shacl.ttl (hunk 13 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_shacl.ttl]: done

File: surface-model/surfacemodel_shacl.ttl (hunk 11 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_shacl.ttl]: same sub category in hdmap but different properties!

File: surface-model/surfacemodel_shacl.ttl (hunk 10 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_shacl.ttl]: same for all specific shacls! let's discuss this in the meeting

File: surface-model/surfacemodel_shacl.ttl (hunk 9 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_shacl.ttl]: add subcategory string part : change to dataType, formatType

File: surface-model/surfacemodel_ontology.ttl (hunk 3 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl]: done! use DataResource

File: surface-model/surfacemodel_ontology.ttl (hunk 2 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl]: use surface-model is comment

File: surface-model/surfacemodel_ontology.ttl (hunk 1 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl:12]: done in ontology file

File: surface-model/surfacemodel_ontology.ttl (hunk 14 above)
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply surface-model/surfacemodel_ontology.ttl]: done

File: surface-model/surfacemodel_shacl.ttl (hunk 10 above)
Comment (MircoNierenz) [reply surface-model/surfacemodel_shacl.ttl]: has different enums for each specific asset data type! 
cannot be reused in this way

File: surface-model/surface-model_ontology.ttl [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+@prefix surface-model: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/surface-model/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .
+
+surface-model: a owl:Ontology ;
+    rdfs:label "ontology definition for surface-model"@en ;
+    dcterms:contributor "Sebastian Tuttas (3DM)" ;
+    owl:versionInfo "0.1" .
+
+surface-model:surfaceModel a owl:Class ;
```
Comment (robertschubert) [inline surface-model/surface-model_ontology.ttl:14]: Classnames in PascalCase
--> SurfaceModel

File: surface-model/surface-model_shacl.ttl [hunk 17]
```diff
@@ -0,0 +1,141 @@
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix surface-model: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/surface-model/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix general: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/general/> .
+@prefix georeference: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/georeference/> .
+
+surface-model:surfaceModelShape a sh:NodeShape ;
```
Comment (robertschubert) [inline surface-model/surface-model_shacl.ttl:8]: Classnames in PascalCase
--> SurfaceModelShape

File: surface-model/surface-model_shacl.ttl [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+            sh:minCount 1 ;
+            sh:node surface-model:QuantityShape ;
+            sh:order 5 ;
+            sh:path surface-model:Quantity ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node georeference:GeoreferenceShape ;
+            sh:order 6 ;
+            sh:path surface-model:Georeference ] ;
+    sh:targetClass surface-model:surface-model .
```
Comment (robertschubert) [inline surface-model/surface-model_shacl.ttl:39]: When ontology is fixed it should be 
`sh:targetClass surface-model:SurfaceModel .`

File: surface-model/surface-model_ontology.ttl [hunk 19]
```diff
@@ -0,0 +1,18 @@
+@prefix dcterms: <http://purl.org/dc/terms/> .
```
Comment (robertschubert) [inline surface-model/surface-model_ontology.ttl:1]: surface-model_instance.json file is missing.

**Review by robertschubert — CHANGES_REQUESTED** [review]

File: surface-model/surface-model_ontology.ttl (hunk 16 above)
Comment (MircoNierenz) [reply surface-model/surface-model_ontology.ttl:14]: fixed

File: surface-model/surface-model_shacl.ttl (hunk 17 above)
Comment (MircoNierenz) [reply surface-model/surface-model_shacl.ttl:8]: fixed

File: surface-model/surface-model_shacl.ttl (hunk 18 above)
Comment (MircoNierenz) [reply surface-model/surface-model_shacl.ttl:39]: fixed

File: surface-model/surface-model_ontology.ttl (hunk 19 above)
Comment (MircoNierenz) [reply surface-model/surface-model_ontology.ttl:1]: done!

**Review by robertschubert — APPROVED** [review]
approved

**Review by MircoNierenz — APPROVED** [review]
ok for me

**Comment by robertschubert** [comment]
resolved merge conflict

**Pull request merged**